MBBSEmu/Extensions/UshortExtensions.cs

```diff
@@ -67,5 +67,20 @@ public static bool Parity(this ushort b)
 
             return setBits != 0 && setBits % 2 == 0;
         }
+
+        /// <summary>
+        ///     Sign extends 16bit -> 32bit
+        /// </summary>
+        /// <param name="b"></param>
+        /// <param name="bitDifference"></param>
+        /// <returns></returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static uint ToUintSignExtended(this ushort b)
+        {
+            if (b.IsNegative())
+                return (uint)(b | 0xFFFF0000);
+
+            return (uint)b;
```

Comment:
can be replaced with a simpler `return (uint)(short)b;`

Reply:
Updated all 3 with wizardry!